utils/chunked_vector.hh

```diff
@@ -372,6 +377,28 @@ chunked_vector<T, max_contiguous_allocation>::chunked_vector(Iterator begin, Ite
     }
 }
 
+template <typename T, size_t max_contiguous_allocation>
+template <std::ranges::range Range>
+requires std::convertible_to<std::ranges::range_value_t<Range>, T>
+chunked_vector<T, max_contiguous_allocation>::chunked_vector(std::from_range_t, Range&& range)
+        : chunked_vector() {
+    if constexpr (std::ranges::forward_range<Range>) {
+        size_t size = std::ranges::distance(range);
+        reserve(size);
+        auto begin = std::ranges::begin(range);
+        for (size_t i = 0; _size < size; ++i) {
+            T* dst = _chunks[i].get();
+            auto now = std::min(size - _size, max_chunk_capacity());
+            begin = std::ranges::uninitialized_copy_n(begin, now, dst, dst + now).in;
+            // Update _size incrementally to let the destructor
+            // know how much data to destroy on exception
+            _size += now;
```

Comment:
If I remember correctly, you had another copy of this exact same code. Would it make sense to create a (static) function that does it once?

Reply:
It's not exactly the same.